.github/workflows/test-e2e-cli.yml

```diff
@@ -35,30 +42,69 @@ jobs:
         if: github.ref == 'refs/heads/main'
         run: echo "TEST_COVERAGE=true" >> $GITHUB_ENV
 
-      - name: Checkout code
+      - name: Checkout
         uses: actions/checkout@v4
 
+      - name: Setup Go, tools and caching
+        uses: ./.github/actions/go-setup
+        with:
+          build-type: e2e-tests-cli
+          without-cache: ${{ inputs.without-cache }}
+          cache-s3-bucket: ${{ vars.CACHE_S3_BUCKET }}
+
       - name: Set TEST_KBC_PROJECTS_FILE entries
         uses: ./.github/actions/export-kbc-projects
         with:
           secrets: ${{ toJSON(secrets) }}
 
-      - name: Install Go
-        uses: actions/setup-go@v5
-        with:
-          go-version: ${{ env.GO_VERSION }}
-          cache: false
-
-      - name: Load Go build cache
-        uses: ./.github/actions/go-cache
-        with:
-          key: e2e-cli
-
-      - name: Load Go tools cache
-        uses: ./.github/actions/go-tools-cache
-
       - name: Run tests
-        run: make tests-cli
+        run: |
+          set -eo pipefail
+          
+          if [ "$RUNNER_OS" == "Windows" ]; then
+            export TEST_BINARY_CLI="${{ runner.temp }}\test-build-cli.exe"
+          else
+            export TEST_BINARY_CLI="${{ runner.temp }}/test-build-cli"
+          fi
+          
+          # Prevent linker warnings on macOs (https://github.com/golang/go/issues/61229#issuecomment-1988965927)
+          if [ "$RUNNER_OS" == "macOS" ]; then
+            export GOFLAGS="-ldflags=-extldflags=-Wl,-ld_classic"
+          fi
+          
+          # Build binary for test without external inputs
+          env -i  \
```

Comment:
@jachym-tousek-keboola Didn't you want this option to be verbosly mentioned as `--ignore-environment`?

Reply:
https://github.com/keboola/keboola-as-code/pull/1818#discussion_r1628915272